lib/mailman/route/conditions.rb

```diff
@@ -4,8 +4,14 @@ class Route
     # Matches against the To addresses of a message.
     class ToCondition < Condition
       def match(message)
-        if !message.to.nil?
+        # this is done for cases where the `to` field is missing but does
+        # have a `Delivered-To` - so we pick from that instead
+        unless message.to.nil? && message.header['Delivered-To'].nil?
           messageto = message.to.is_a?(Array) ? message.to : [message.to]
```

Comment:
we should fallback to `Delivered-To` only when `message.to` is nil


so first check if message.to is present, do the routing based on it, if message.to is nil then fallback to `Delivered-To`

Reply:
yeah - but first we check if either of them has a value - otherwise its pointless to extract the attribute.